modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/AwsS3ServiceImplTests.java

```diff
@@ -229,4 +233,48 @@ private void assertEndpoint(Settings repositorySettings, Settings settings, Stri
         assertThat(clientSettings.endpoint, is(expectedEndpoint));
     }
 
+    public void testLoggingCredentialsProviderCatchesErrors() {
+        var mockProvider = Mockito.mock(AWSCredentialsProvider.class);
+        String mockProviderErrorMessage = "mockProvider failed to generate credentials";
+        Mockito.when(mockProvider.getCredentials()).thenThrow(new IllegalStateException(mockProviderErrorMessage));
+        var mockLogger = Mockito.mock(Logger.class);
+
+        var credentialsProvider = new S3Service.ErrorLoggingCredentialsProvider(mockProvider, mockLogger);
+        try {
+            credentialsProvider.getCredentials();
+            fail("Shouldn't successfully get credentials");
```

Comment:
Can we use expectThrows() instead?

Reply:
Thank you! I keep forgetting about the assertion helpers from the Lucene test utils framework.